crates/ty_python_semantic/src/types/subclass_of.rs

```diff
@@ -73,6 +73,17 @@ impl<'db> SubclassOfType<'db> {
         !self.is_dynamic()
     }
 
+    pub(super) fn top_materialization(self, db: &'db dyn Db) -> Type<'db> {
```

Comment:
Doesn't this still need to have a variance parameter? Because the subclass-of type might be itself in a contravariant position, in which case we should be materializing to `Never` instead.

Reply:
Can you give me an example? I'm not sure I follow here.

Do you mean that the top materialization of `Callable[[type[Any]], type[Any]]` should be `Callable[[type[Never]], type[object]]` and not `Callable[[type[object]], type[object]]`?

Reply:
Yes, exactly!

Reply:
Updated this. The materialization happens like so:
* For covariant, `type[Any]` -> `type[object]` (which is `type`)
* For contravariant, `type[Any]` -> `type[Never]` (which is `Never`)
* For invariant, `type[Any]` -> `type[T] (T: type)`